internal/rootcoord/util.go

```diff
@@ -135,3 +135,15 @@ func getTravelTs(req TimeTravelRequest) Timestamp {
 func isMaxTs(ts Timestamp) bool {
 	return ts == typeutil.MaxTimestamp
 }
+
+func getNeedChanNum(setNum int, chanMap map[typeutil.UniqueID][]string) int {
+	var ret int
+
+	for _, chanNames := range chanMap {
+		ret += len(chanNames)
+	}
+	if setNum > ret {
+		ret = setNum
+	}
+	return ret
```

Comment:
case 
chanMap = map[int64]channels {
1: {"dml_1", "dml_2"},
2: {"dml_2", "dml_3"},
}
needed number shall be 3, not 4